clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -201,7 +201,8 @@
     public static final String MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION = "max.in.flight.requests.per.connection";
     private static final String MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION_DOC = "The maximum number of unacknowledged requests the client will send on a single connection before blocking."
                                                                             + " Note that if this setting is set to be greater than 1 and there are failed sends, there is a risk of"
-                                                                            + " message re-ordering due to retries (i.e., if retries are enabled).";
+                                                                            + " message re-ordering due to retries (i.e., if retries are enabled). With an idempotent producer, this"
+                                                                            + " can be up to 5 and still provide ordering guarantees.";
```

Comment:
The `enable.idempotence` will be true by default in 3.0, so we should perhaps have the "up to 5" thing first.

Reply:
@ijuma done